src/pixel_formats.rs

```diff
@@ -0,0 +1,308 @@
+                color_spec: ColorSpecification::ColorFormat(ColorFormat {
+                    red_max: rgb_888::MAX_VALUE,
+                    green_max: rgb_888::MAX_VALUE,
+                    blue_max: rgb_888::MAX_VALUE,
+                    red_shift: rgb_888::BITS_PER_COLOR * 1,
+                    green_shift: rgb_888::BITS_PER_COLOR * 2,
+                    blue_shift: rgb_888::BITS_PER_COLOR * 3,
+                }),
+            }),
+
+            v => Err(anyhow!("unknown fourcc: {}", v)),
+        }
+    }
+}
+
+/// Utility functions for 32-bit RGB pixel formats, with 8-bits used per color.
+pub mod rgb_888 {
+    use crate::rfb::{ColorSpecification, PixelFormat};
+
+    pub const BYTES_PER_PIXEL: usize = 4;
+    pub const BITS_PER_PIXEL: u8 = 32;
+
+    /// Number of bits used for color in a pixel
+    pub const DEPTH: u8 = 24;
+
+    /// Number of bits used for a single color value
+    pub const BITS_PER_COLOR: u8 = 8;
+
+    /// Max value for a given color
+    pub const MAX_VALUE: u16 = 255;
+
+    /// Returns true if a shift as specified in a pixel format is valid for rgb888 formats.
+    pub fn valid_shift(shift: u8) -> bool {
+        shift == 0 || shift == 8 || shift == 16 || shift == 24
+    }
+
+    /// Returns the byte index into a 4-byte pixel vector for a given color shift, accounting for endianness.
+    pub fn color_shift_to_index(shift: u8, big_endian: bool) -> usize {
+        assert!(valid_shift(shift));
+
+        if big_endian {
+            ((DEPTH - shift) / BITS_PER_COLOR) as usize
+        } else {
+            (shift / BITS_PER_COLOR) as usize
+        }
+    }
+
+    /// Returns the index of the unused byte (the only byte not representing R, G, or B).
+    pub fn unused_index(r: usize, g: usize, b: usize) -> usize {
+        (3 + 2 + 1) - r - g - b
+    }
+
+    /// Given a set of red/green/blue shifts from a pixel format and its endianness, determine
+    /// which byte index in a 4-byte vector representing a pixel maps to which color.
+    ///
+    /// For example, for the shifts red=0, green=8, blue=16, and a little-endian format, the
+    /// indices would be red=0, green=1, blue=2, and x=3.
+    pub fn rgbx_index(
+        red_shift: u8,
+        green_shift: u8,
+        blue_shift: u8,
+        big_endian: bool,
+    ) -> (usize, usize, usize, usize) {
+        let r = color_shift_to_index(red_shift, big_endian);
+        let g = color_shift_to_index(green_shift, big_endian);
+        let b = color_shift_to_index(blue_shift, big_endian);
+        let x = unused_index(r, g, b);
+
+        (r, g, b, x)
+    }
+
+    /// Translate between RGB888 formats. The input and output format must both be RGB888.
+    pub fn transform(pixels: &Vec<u8>, input: &PixelFormat, output: &PixelFormat) -> Vec<u8> {
+        assert!(input.is_rgb_888());
+        assert!(output.is_rgb_888());
+
+        let mut buf = Vec::with_capacity(pixels.len());
+        buf.resize(pixels.len(), 0x0u8);
```

Comment:
```suggestion
        let mut buf = vec![0; pixels.len()];
```

Reply:
hm, yeah. I had a reason for structuring it this way initially but I forget what it was. The way you suggested is simpler.

done